### PR TITLE
Feature/user dashboard

### PR DESCRIPTION
**Pull request by nachtiomer**

### CODE REVIEW

File: src/components/User/index.js [hunk 1]
```diff
@@ -1,14 +1,18 @@
-import React, { useState } from "react";
-import NewMission from "./Mission/NewMission";
-import UserDashboard from "./UserDashboard";
-import FadeIn from "react-fade-in";
+import React from 'react';
```
Comment (amitaiblevy) [inline src/components/User/index.js:1]: Why does this component exist? Couldn't the div and FadeIn go in UserDashboard

File: src/components/User/UserDashboard/index.js [hunk 2]
```diff
@@ -1,135 +1,132 @@
...
+const getUserPages = user => Pages.filter(page => page.availableTo.includes(user.type)).sort((a, b) => a.index - b.index);
+
+const UserDashboard = props => {
+  const globalState = useGlobalState();
+  const [selectedPage, setSelectedPage] = useState(undefined);
+
+  const availablePages = React.useMemo(() => getUserPages(globalState.loggedInUser), [globalState.loggedInUser]);
+  const ActivePage = React.useMemo(() => availablePages.find(page => page.name === selectedPage), [availablePages, selectedPage]);
+  
+  React.useEffect(() => {
```
Comment (amitaiblevy) [inline src/components/User/UserDashboard/index.js:63]: import React, {useEffect} from 'react'

File: src/components/User/UserDashboard/index.js [hunk 3]
```diff
@@ -1,135 +1,132 @@
...
+    }).catch(console.log);
+  }, [globalState.wrapWithLoading, globalState.loggedInUser, globalState.setMissions]);
+
+  const reloadTasks = React.useCallback(async () => {
+    const {loggedInUser, wrapWithLoading, setTasks} = globalState;
+    wrapWithLoading(async () => {
+      const tasks = await fetchUserTasks(loggedInUser._id, loggedInUser.type);
+      setTasks(tasks);
+    });
+  }, [globalState.wrapWithLoading, globalState.loggedInUser, globalState.setTasks]);
```
Comment (amitaiblevy) [inline src/components/User/UserDashboard/index.js:91]: If you put in here globalState.setTasks and globalState.wrapWithLoading, the useCallback is useless because those parameters are reset every time. You can remove them from the useCallback or remove the useCallback

File: src/components/User/UserDashboard/index.js [hunk 4]
```diff
@@ -1,135 +1,132 @@
...
+  return (
+    <div id="userDashboard">
+      <div id="pagesButtonsBar" >
+        {availablePages
+          .filter(page => page.type === "button")
+          .map(page => (
+            <TabButton
+              key={page.name}
+              page={page}
+              onClick={OnClickHandler}
```
Comment (amitaiblevy) [inline src/components/User/UserDashboard/index.js:102]: Capital?

File: src/utils/api.js [hunk 5]
```diff
@@ -11,6 +11,11 @@ export const fetchUsers = async () => {
   return await response.json();
 };
 
+export const fetchMissions = async () => {
```
Comment (amitaiblevy) [inline src/utils/api.js:14]: Why is this here if its not used? I realize it't in the server but I don't want it used, so I'd rather erase it from the server

File: src/components/User/index.js (hunk 1 above)
Comment (nachtiomer) [reply src/components/User/index.js:1]: It's just a wrapper for the user section,
right now there's only dashboard but when we'll start using react router as it should be used it can make our lives much easier...

File: src/components/User/UserDashboard/index.js (hunk 3 above)
Comment (nachtiomer) [reply src/components/User/UserDashboard/index.js:91]: you want to cache the function as long as you give it as a prop to children components,
not using usecallback on prop functions causing children rerendering even if it's not needed.
I'll use the globalState functions to remove the callback re-caching dependency.
